src/components/Image/Image.tsx

```diff
@@ -102,71 +59,66 @@ export class Image extends React.Component<ImageProps> {
       loadingBehavior,
       ...imageProps
     } = this.props;
-    const { isLoaded } = this.state;
+    const { isLoaded, boundingRectDimensions } = this.state;
 
     const isAbsoluteUrl = src && src.match('^https?://');
 
     const aspectRatioAsNumber =
       typeof aspectRatio === 'number'
         ? aspectRatio
         : AspectRatioOptions[aspectRatio];
-    const sourceDimensions = { width, height };
-
-    const containerDimensions = {
-      width:
-        !width && height && aspectRatioAsNumber
-          ? height * aspectRatioAsNumber
-          : width,
-      height:
-        !height && width && aspectRatioAsNumber
-          ? width / aspectRatioAsNumber
-          : height,
+
+    // Taking the dimensions from props or from its bounding rectangle in case they're missing
+    const sourceDimensions = {
+      width: width || boundingRectDimensions?.width,
+      height: height || boundingRectDimensions?.height,
     };
 
+    // Calculating the dimensions considering the given values and aspect ratio
+    const calculatedDimensions = calculateDimensions({
+      ...sourceDimensions,
+      aspectRatio: aspectRatioAsNumber,
+    });
+
     const hasLoadingBehavior = loadingBehavior === 'blur';
 
     return (
       <div
+        ref={this.containerRef}
         className={st(
           classes.root,
-          {
-            resize,
-            ...(hasLoadingBehavior && { preload: !isLoaded, loaded: isLoaded }),
-          },
           resize === ResizeOptions.cover ? classes.cover : classes.contain,
           className,
         )}
         data-hook={this.props['data-hook']}
       >
         {isAbsoluteUrl ? (
           <CoreImage
-            {...imageProps}
-            nativeProps={{ ...containerDimensions }}
             src={src}
-            className={classes.absoluteImage}
+            className={st(
+              classes.absoluteImage,
+              hasLoadingBehavior &&
+                (isLoaded ? classes.loaded : classes.preload),
+            )}
```

Comment:
`st` isn't needed here, but only for the root (or when using states).
You can use `classnames` for this:
```suggestion
            className={classnames(
              classes.absoluteImage,
              { 
                [classes.loaded]: hasLoadingBehavior && isLoaded,
                [classes.preload]: hasLoadingBehavior && !isLoaded, 
              })}
```